### PR TITLE
Add datalayer to GTM in login 

### LINKED ISSUE #1190: Extra GTM <noscript> added when navigating between pages
**Steps to recreate:**
1. Sign in to your account 
2. Go to homepage on staging https://staging.fightpandemics.work/
3. Click on **any** links eg `About Us`, `Help Board`, `Request Help`, `Offer Help`...
4. Observe that an extra <noscript> tag is added 

It's a GTM related element
```
      <iframe src="https://www.googletagmanager.com/ns.html?id=undefined&gtm_auth=&gtm_preview=&gtm_cookies_win=x"
        height="0" width="0" style="display:none;visibility:hidden" id="tag-manager"></iframe>
```

Screen Shot 2020-07-16 at 12 52 21 PM

Looks like every time we call `TagManager.initialize` it inserts a new one, https://github.com/alinemorelli/react-gtm/blob/master/src/TagManager.js#L43

### PR DESCRIPTION
**Pull request by nr431207**
### What does this pull request aim to achieve? 💯
Moving adding the datalayer  to Login page from Navbar
fixes #1190 
_Please be concise and link any related issue(s):_

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](https://github.com/FightPandemics/FightPandemics/blob/master/README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`** (need to have cloned the repo not forked).
- [x] This branch is rebased/merged with the **latest master**.
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".

### CODE REVIEW

**Comment by nr431207** [comment]
I tested on my end and with the changes @idjevm added it is now adding userId for non-logged in users as well, thanks @idjevm !

**Comment by idjevm** [comment]
Yeah atm, I am just adding -1 for non logged in users, I know we are tracking non logged in users by their IP with GA, so we don't have to worry about that in this case.

File: client/src/actions/authActions.js [hunk 1]
```diff
@@ -7,8 +7,18 @@ import {
...
 const GET_CURRENT_USER_ENDPOINT = "/api/users/current";
 
+const getTagManagerArgs = (userId) => {
+  return {
+    dataLayer: {
+      userId: userId,
+    },
+  };
+};
+
```
Comment (jyclam) [inline client/src/actions/authActions.js:21]: Can just use implicit return here and get rid of one layer of brackets. 
```
const getTagManagerArgs = (userId) => {
    dataLayer: {
      userId: userId,
    },
};
```

Are you still planning to change the name as we discussed?

File: client/src/actions/authActions.js [hunk 2]
```diff
@@ -17,6 +27,7 @@ export const initAuth = () => {
     dispatch({ type: SET_AUTH_LOADING, payload: true });
     try {
       const { data: user } = await axios.get(GET_CURRENT_USER_ENDPOINT);
+      TagManager.dataLayer(getTagManagerArgs(user?.id));
```
Comment (jyclam) [inline client/src/actions/authActions.js:30]: Don't think you need safe nav here

File: client/src/pages/Login.js [hunk 3]
```diff
@@ -210,6 +211,14 @@ const VisibilityButton = ({ onClick, type }) => {
...
 };
 
+const getTagManagerArgs = (userId) => {
+  return {
+    dataLayer: {
+      userId: userId,
+    },
+  };
+};
+
```
Comment (jyclam) [inline client/src/pages/Login.js:221]: Same comment as above, can use implicit return and rename

File: client/src/pages/Login.js [hunk 4]
```diff
@@ -249,7 +264,7 @@ const Login = ({ isLoginForm, forgotPassword }) => {
 
     try {
       const res = await axios.post("/api/auth/login", formData);
-
+      TagManager.dataLayer(getTagManagerArgs(res.data?.user?.id));
```
Comment (jyclam) [inline client/src/pages/Login.js:267]: Don't need safe nav here

Comment (idjevm) [reply client/src/pages/Login.js:267]: Ok! Removed!

File: client/src/actions/authActions.js (hunk 1 above)
Comment (idjevm) [reply client/src/actions/authActions.js:21]: No naming, just implicit sending the userId :)

File: client/src/actions/authActions.js (hunk 2 above)
Comment (idjevm) [reply client/src/actions/authActions.js:30]: Ok removed :D

**Pull request merged**